component.go

```diff
@@ -58,9 +60,14 @@ type portHandler struct {
 	onClose reflect.Value
 }
 
+// inputCounter keeps track of total additions and decrements have occured
+// on inputsClose
+
+var inputCounter = 0
+
 // RunProc runs event handling loop on component ports.
 // It returns true on success or panics with error message and returns false on error.
-func RunProc(c interface{}) bool {
+func RunProc(c interface{}, nump int) bool {
```

Comment:
`nump` is not used in the code. I assume it has been superseded by `poolMode` and should be removed.